{{ cookiecutter.app_name }}/setup.py

```diff
@@ -0,0 +1,87 @@
+"""
+Install {{ cookiecutter.app_name }} via setuptools
+"""
+import sys
+
+from setuptools import setup, find_packages
+from setuptools.command.test import test as testcommand
+
+with open('test_requirements.txt') as test_reqs:
+    tests_require = test_reqs.readlines()  # pylint: disable=invalid-name
+
+
+class PyTest(testcommand):
+    """PyTest class to enable running `python setup.py test`"""
+    user_options = testcommand.user_options[:]
+    user_options += [
+        ('coverage', 'C', 'Produce a coverage report for {{ cookiecutter.app_name }}'),
+        ('pep8', 'P', 'Produce a pep8 report for {{ cookiecutter.app_name }}'),
+        ('pylint', 'l', 'Produce a pylint report for {{ cookiecutter.app_name }}'),
+    ]
+    coverage = None
+    pep8 = None
+    lint = None
+    test_suite = False
+    test_args = []
+
+    def initialize_options(self):
+        testcommand.initialize_options(self)
+
+    def finalize_options(self):
+        testcommand.finalize_options(self)
+        self.test_suite = True
+        self.test_args = []
+        if self.coverage:
+            self.test_args.append('--cov')
+            self.test_args.append('{{ cookiecutter.app_name }}')
```

Comment:
Might be clearer as `self.test_args.extend(['--cov', '{{ cookiecutter.app_name }}'])`.